go/teams/loader.go

```diff
@@ -909,9 +907,15 @@ func (l *TeamLoader) load2InnerLockedRetry(ctx context.Context, arg load2ArgT) (
 	}
 
 	if hd := hiddenPackage.ChainData(); hd != nil {
+		hd.NeedRotate = needHiddenRotate
 		load2res.hidden = hd
 	}
 
+	if !arg.fromBoxAudit && needHiddenRotate {
+		mctx.Info("Scheduling box audit since we needed a rotation upon load (and we're not being directly called from the box audit)")
```

Comment:
Make this a `Debug`?

Also, as discussed this will recurse up the parent chain in the worst case causing quadratic audits. So say A.B.C.D.E, and all teams need hidden rotations. Box auditing E will load A, B, C, and D and then box audit them since this flag isn't passed in through `MapTeamAncestors`. Then the box audit for D will cause a box audit for A, B, C, and so on.

Reply:
also box audits succeed without checking for readers. if that's not ok we could pass in an option to make that not ok. I'm guessing only writers will attempt to rotate team hidden though?